CMS-Enterprise/easi-app
# [NOREF] Upgrade ESLint & Prettier

# NOREF

## Description

> [!tip]
> Want to see a similar change on MINT? Check out https://github.com/CMSgov/mint-app/pull/1326!

> [!important]
> This PR _did_ remove a bunch of eslint plugins from `package.json`
> This is because we were defining these imports ourselves, and often we had to have manual `resolutions` specified to make sure that we only ever downloaded 1 version of a plugin.
> However -- all of the `eslint-config-*` packages _already_ provide these plugins, and we can still safely use and reference them in our configs because of this!

- Updates ESLint from V7 to V8
  - The upgrade to V9 is a bit tougher, so just doing this version for now!
- Updates Prettier from V2 to V3
  - This _does_ result in a number of automatic changes to a bunch of files, but should be consistent with new defaults introduced in Prettier V3!

## How to test this change

- Review changes in `package.json`
- Ensure `scripts/dev lint` works locally (after running `yarn install --immutable`!)
- (Roughly) Review automatic changes as a result of `prettier` major version bump
  - These should only be formatting changes -- nothing here should affect any logic!

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
